### PR TITLE
Update pricing page link

### PR DESCRIPTION
**Pull request by kimberleyamackenzie**
## Checklist:

https://github.com/orgs/pulumi/projects/33/views/3?filterQuery=assignee%3Akimberleyamackenzie&pane=issue&itemId=24292771

Fixes https://github.com/pulumi/pulumi-hugo/issues/2624, which was originally from https://github.com/pulumi/pulumi-service/issues/8473.  

This updated link points to sign up, and then a new org dialog should open automatically after signing up (as per existing behavior).

- [x] I have manually confirmed that all new links work.
- [x] I have reviewed the [style guide](../STYLE-GUIDE.md).
- [ ] If blogging, I have reviewed the [blogging guide](../BLOGGING.md).

### CODE REVIEW

**Comment by kimberleyamackenzie** [comment]
@SaraDPH Since you originally wrote this issue, wanted to triple check that this change is still wanted + looks good to you.

**Review by susanev — APPROVED** [review]
looks good to me, but we should wait on Sara's approval

**Review by SaraDPH — APPROVED** [review]

**Pull request merged**
